### DISCUSSION TITLE
Changing Layout - FlexBox is your Friend

### DISCUSSION THREAD

**Discussion by giscus** [LiaScript/LiaScript#188] [General]
# Changing Layout - FlexBox is your Friend

How to change the layout of certain parts of your LiaScript course easily with FlexBox

https://liascript.github.io/blog/changing-layout-flexbox-is-your-friend/

**Comment by michaelkowalczyk** [comment]
Flex it up, nice addition!

**Comment by tilman-schieber** [comment]
I have used these suggestions in my course and they are a nice way to position images and other media elements. But using the comment-syntax e.g. `<!-- class="flex-child"-->` has some confusing effects. If I use

```
<div class="flex-container">

<!-- class="flex-child" -->
![img](image.png)

<!-- class="flex-child" -->
lorem ipsum

</div>
```

The image is interpreted as a "beginning of paragraph"-image for some reason and is scaled to 50%.

see [here](https://liascript.github.io/course/?data:text/plain;charset=utf-8;Content-Encoding=gzip;base64,H4sIAAAAAAAAA31SzW4TMRC+R8o7TMMFquaHH3HYFkQRiAtXTghVE3uSHclru/Y4Sal4Ft6CF+iLMbNpBT3AZbW2Z76/mYuT+Xw6eVflJtB0Mp0sNoEOc5eiIEcqcDudAAxYthznknIHL1f5cG6XnmsOeNOBdYw3Y6vnQk44xQ5K2v+53xfUbvuew/IULkNI+wrSE7DQoH9pfASsEIk8eThdWjMG3sb5WNNBlULi+hF1a3gvTM2I532FdZIe+lT4u+kPgNHDjoqw00PN6DhuYU2yJ4r3tEby4y/jPQcPt/Agu4Pn5/cFmhJFP53M52/t8ATeW2lsBThCj2GtaX0qd7/uftKx/MLzDlzAWt/MHqc6U4QLDf7xq1HPRvSTr4b97Wkvkmu3XGrOjvoU/MKl5avV6vB6tYLZ1eV6rWVNPT3vrmDP5R9ywOv8qAqFILNno7T/kH9OhQbgXNsAPoVUoLIADiRnoA6qTpdEadBz5jpGSoH1serMfALiVofkQePNydQ49qwqBZpAwLXCA8kRmnS1thFtyNcNF/BFgCIPig0D289OjzicwXVj3Yuk828e6EDFsaBtGbQQcHDpiGxFXNmYRkjOWgyEKnxQTeloQKlkAR8MEpvoApamSo5eNbxCuVCvs6aixvVil0LLSkcqR50C1UrgOISHhNRQg03bMopOQAVBxqKHVhbw8eAoCzWLUTNIziE5rXMts0exDnWRS2JP0VK0pJTUtZDRfEPabNgxgqdKxV6HFEwGWkCscdT7XNuwGIe71MXTQf4GeVOS0tsDAAA=)

Another gripe of mine is that for any general purpose layout is is very unlikely that a flex-child consists of only one markdown block element. Generally a column in a document contains more than one paragraph of text or one bullet point list. So I have to fall back to normal divs anyway. That's of course fine, but it also behaves a bit finicky, with empty lines required before `</div>` commands and so on.

I know that's not the most constructive feedback as I have no clear suggestion how to improve on this. Mixing html and Markdown is challenging in most Markdown dialects. 

I personally would love some syntax like
```
<!-- @div(flex-child) -->
multiple

paragraphs
<!-- @end-div -->
```
but I was not able to get this to work with any kind of macro. If I define a macro to contain html tags, they are output as plain text.

**Comment by andre-dietrich** [comment]
Thanks for the issue, this was a styling/elm misinterpretation ... If there is an image and a block-comment to the head is added, then wrongly it was placed into a <p> which resulted in that odd half-size image width... This is now placed into a div with the block-attributes, such that the image or any other multimedia content will be spaced correctly ...

https://github.com/LiaScript/LiaScript/commit/f2aad7ee975fe962cfe9a1e0ba5590309a7d16e5

to deal with this at the moment, you can also style a div and add the multimedia-element into it ... here is a little example, I will also update this int the blog:

[LiveEditor example](https://liascript.github.io/LiveEditor/?/show/code/H4sIAAAAAAAAA+1V227bRhB951eMXRSNA5O05StkO3bTFkUAGy3aoGlhGMZyOSE3WnKZvUhWAv97z9ISG0tugTwUeYkebO2cnZmzM3NGpxtpmlw4P9ecZG8136XStF6oli19TAifUrlOi/mYInrSm/p7Myu6McW/J5Q/p++1NjNHvmZSnht8Mz1IwlHLXHJJz/PeW2hVtWl/aUzOW/ayfohbxYi7Ox3S3CdLOrXS5fanh3SXPvYUcPeE7h9BowEarUJ7A7S3Cu0P0P4qdDBAB6vQ4QAdrkJHA3S0Ch0P0HGEkouGSyXoWSNQUlX6Gpl2UIKtRfkfVWHFkO6um0brpr110/666WDddLhuOlo3HS+YLicjtnDn234orsSEH2aiv4pZmuIII4WuN78NWlP/bDItOa0aKnmqJLwW0xI/jbCValOrqtqPaacP/Rs3ZroI3gOLW0u/e4zQBbdlkqYvkuQbuhRzEzxyIdVV0F71dU+SU8fSKySXWjh3tvlYA5vwPQXrx+hD7Tepl83ZZgNui9aNYutOloQL471pxpTGmY6hNs6vb57V3ndunOez2SyLnELBmTRNPhMQwvn0TNe/v/vr9VH3ZjLbQvYc6SMLCPUJFqOBRSHkpLImtOWYdCxIZcUcVP4htxfJbVIsyKWx3JDqXGjIlMaSaNhvE94dy8E+QP6iVJ1yUrUVsVY+oz/UVDTBEVfsfJS1cppQ085Y75VHlI5b4VURXEa/1gLKtoJ0kD7EHSBRFtxz1C0h9uS46dhm9CN2gnIO/e85vA/Cx/iK2Bp3QjJYUSgfLNXxP6rAVKgC/Q1NRr9YqRZA6yOjQou2VB5PU2Y42Nq0MlK7hKM1cQ+9D6AQYzNN1ZQtOPVJwSEUUOXg8qqV3HnjCLRqZLVsEdEFhxdjQYJNhyCOkX7KGg+9U7FSzmitJHiXqs3+dZL2PqOH+08O2HK+3tRzQm3J850vjJlEBsR3IOnwvCRJ6fqabm7oNUTTCRSe0I5GlBCiNQ1Z+G4TXiTaii02NpYzu2xw+0nImmLc6FajqiAArlQgaytVfH1jWmSFDAVaK9Fm0kaKKLDsUfZJa2aaywq/FuAGAqrn2lmWyoQoa1Ra+e8wLRbroCQhpbEoY6XnD5H+RKSXwUGlzp1v/F9C2fhEsKHTRpTZTE0elkdmbJXHUxdPOTSM17vc16Ep8lE+EvnPwXOLYatuX6pC8+0VlsqH2x9MN8/edVW+uzPa7+7S/7619VWsX0asT8/R7mfM0ejrwv3CPcwXv+4vkr8BUyPsGekKAAA=)
